tools/packager/CMakeLists.txt

```diff
@@ -36,7 +36,11 @@ FIND_PYTHON_PKG("jsonschema" SEND_ERROR)
 set(PKG_COMMON_FILES
     afu.py gbs.py utils.py metadata/constants.py metadata/__init__.py metadata/metadata.py schema/afu_schema_v01.json)
 
-set(PKG_PYTHON_FILES packager.py ${PKG_COMMON_FILES} README)
+################# generate the py file for the packager ########################
+configure_file(packager.py.in packager.py @ONLY)
+
+set(${CMAKE_CURRENT_BINARY_DIR}/packager.py ${CMAKE_CURRENT_BINARY_DIR}/packager.py packager.py)
```

Comment:
what is this line doing?

Reply:
It's a cmake syntax - only replace variables in the input that are surrounded by '@', like @CMAKE_VAR@

Reply:
I'm talking about line 42 (The answer to the ultimate question line life...)

Reply:
Line 42 is making a list, which is recognized by the CREATE_PYTHON_EXE macro and handled properly.  It zips the file with var[0] and names the module var[1].

Reply:
I thought line 43 was making that list (`PKG_PYTHON_FILES`) which gets passed in to the macro in line 44 (`${PKG_PYTHON_FILES}`)

Reply:
I added this ability recently to the zipper. The zipper still takes a list of files to zip. In addition, entries in the list can be tuples, implemented as a list with 2 entries. The first entry in the tuple is the file to include in the zipped image. The second entry in the tuple is the name to call the file in the zipfile. This allows us to write a CMake rule that modifies a source Python file and then embeds the modified Python file in the target zipfile.

Reply:
The code is here: https://github.com/OPAE/opae-sdk/blob/0da43d40f331bb101c9dc9762491eb259aff9897/cmake/modules/packaging.cmake#L108

Reply:
Maybe I'm missing something but this line (42):
`set(${CMAKE_CURRENT_BINARY_DIR}/packager.py ${CMAKE_CURRENT_BINARY_DIR}/packager.py packager.py)`
doesn't seem to be doing anything useful.

According to cmake documentation:
`set(<variable> <value>
    [[CACHE <type> <docstring> [FORCE]] | PARENT_SCOPE])`

and line 42:
`<variable>` is `${CMAKE_CURRENT_BINARY_DIR}/packager.py`
`<value>` is `${CMAKE_CURRENT_BINARY_DIR}/packager.py packager.py`

Reply:
I don't think you're missing anything. The concept is correct, the implementation at line 42 is wrong. I bet it's creating a variable with a funny name, which is why it might work.

Reply:
I was only concerned with line 42 and nothing else. It seems like it's wrong and not necessary.  
I suggest we remove it. Doesn't the macro work without it? It's not in the current (unpatched) version of this file.

Reply:
Line 42 could say

set(FOO ${PATH}/packager.py packager.py)

and we can replace the ${PATH}/packager.py with FOO (no ${}) and it should also work.

The idea is to create a variable that is defined that contains a list, the first of which is the file name and the second is the module name defined within the file.  This causes the logic in CREATE_PYTHON_EXE to add the correct parameters to the zipfile.write() method.

Reply:
Doesn't line 43 already to this? It creates the list of files passed in as the third argument to CREATE_PYTHON_EXE.
The first two arguments to CREATE_PYTHON_EXE are the "executable" name and the name of the main module (which happen to both be `packager`).

Reply:
The argument in the *old* version passes a SINGLE list to the CREATE_PYTHON_EXE macro, where my modification passes a list, of which one of the members of the list is another list (file path and module name).  To bastardize cmake/python:

OLD:
CREATE_PYTHON_EXE([packager, packager, afu.py, gbs.py, ...])

NEW:
CREATE_PYTHON_EXE([packager, packager, [/build/packager.py, packager], afu.py, gbs.py, ...])

The macro pulls apart the [/build/packager.py, packager] list into file and module name.  It is unfortunate that I named the list variable '${CMAKE_CURRENT_BINARY_DIR}/packager.py'.  Naming it something like PY_FILE_MODULE probably would make it more clear, like this:

set(PY_FILE_MODULE ${CMAKE_CURRENT_BINARY_DIR}/packager.py packager.py)
set(PKG_PYTHON_FILES PY_FILE_MODULE ${PKG_COMMON_FILES} README)
CREATE_PYTHON_EXE(packager packager ${PKG_PYTHON_FILES})

Clear(er)?

Reply:
Yes.